promql/query_logger.go

```diff
@@ -205,11 +211,17 @@ func (tracker ActiveQueryTracker) GetMaxConcurrent() int {
 }
 
 func (tracker ActiveQueryTracker) Delete(insertIndex int) {
+	if !tracker.isRunning.Load() {
+		return
+	}
 	copy(tracker.mmappedFile[insertIndex:], strings.Repeat("\x00", entrySize))
 	tracker.getNextIndex <- insertIndex
 }
 
 func (tracker ActiveQueryTracker) Insert(ctx context.Context, query string) (int, error) {
+	if !tracker.isRunning.Load() {
+		return 0, errors.New("ActiveQueryTracker is stopped")
```

Comment:
Not sure if returning an error here is the correct thing to do as it will result in failed rule evaluations and logged errors, but that's misleading since the query itself is fine, we just don't track it because tracker is already closed